src/shims/fs.rs

```diff
@@ -261,6 +261,83 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(result)
     }
 
+    fn stat(
+        &mut self,
+        path_op: OpTy<'tcx, Tag>,
+        buf_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        if this.tcx.sess.target.target.target_os.to_lowercase() != "macos" {
+            throw_unsup_format!("The `stat` shim is only only available for `macos` targets.")
+        }
+
+        let path_scalar = this.read_scalar(path_op)?.not_undef()?;
+        let path: PathBuf = this.read_os_str_from_c_str(path_scalar)?.into();
+
+        let buf = this.deref_operand(buf_op)?;
+
+        let status = match FileStatus::new(this, path, false)? {
```

Comment:
So this never follows symlinks? What happens with `symlink_metadata` on macOS?

Looks like we should have a test for `symlink_metadata`, too.

Reply:
Ah you already have a FIXME for tests. So, fine to leave that for later, but the hard-coded `false` here still strikes me as odd.

Reply:
Ah looks like that's a different syscall, `lstat64`. So please just add a comment saying that.